sqld/src/hrana/ws/handshake.rs

```diff
@@ -14,14 +19,29 @@ pub enum WebSocket {
     Upgraded(tokio_tungstenite::WebSocketStream<hyper::upgrade::Upgraded>),
 }
 
-pub async fn handshake_tcp(socket: tokio::net::TcpStream) -> Result<(WebSocket, Version)> {
+fn extract_namespace<B>(req: &Request<B>) -> anyhow::Result<Bytes> {
+    let host = req.headers().get("host").context("missing host header")?;
```

Comment:
I think we should be avoiding to use context and convert this directly into its error type that is well defined and we can then look at how it implements IntoResponse to understand what status/message we will reply with.